### ISSUE TITLE
Parsing error on `<`

### ISSUE THREAD

**Issue by fsbdev** [babel/babel#2083]
Parsing the code in the below comment results in the following error:
`Unexpected token (9:20)`
which points to this line 
`switch ((typeA << 4) | typeB) {`
but this should be valid.

**Comment by sebmck** [comment]
Minimal reproducible example:

```javascript
class Foo {
  foo() {
    switch (1) {
      case (MatrixType.IsScaling | MatrixType.IsTranslation):
    }
  }

  bar() {
    switch ((typeA << 4) | typeB) {}
  }
}

```

It's an issue with the lookahead that's used for arrow function return type annotations.

**Comment by fsbdev** [comment]
Thanks for that, I'll remove my gist

**Comment by sebmck** [comment]
Fixed as of 5.8.9, thanks :blush:

**Comment by jnwng** [comment]
thanks @sebmck!
